src/odbc/src/statement.cpp

```diff
@@ -976,22 +990,88 @@ SqlResult::Type Statement::InternalFreeResources(int16_t option) {
 }
 
 void Statement::Close() {
-  IGNITE_ODBC_API_CALL(InternalClose());
+  IGNITE_ODBC_API_CALL(InternalClose(false));
 }
 
-SqlResult::Type Statement::InternalClose() {
+SqlResult::Type Statement::InternalClose(bool ignoreErrors) {
   if (!currentQuery.get())
     return SqlResult::AI_SUCCESS;
 
-  if (!currentQuery->DataAvailable()) {
+  if (!ignoreErrors && !currentQuery->DataAvailable()) {
     AddStatusRecord(SqlState::S24000_INVALID_CURSOR_STATE,
                     "No cursor was open");
     return SqlResult::AI_ERROR;
   }
 
   SqlResult::Type result = currentQuery->Close();
```

Comment:
Is it safe to call this if DataAvailable() was false? In the previous version this would throw an error first, but now the error can get skipped.

Reply:
Changed so `currentQuery->Close()` cannot be called after `currentQuery->DataAvailable()` is `false`. `ignoreErrors` now does what it sounds like, causes `SqlResult::AI_SUCCESS` to be returned no matter what and does not alter the logic of `InternalClose`.